### PR TITLE
Get version without deprecation warnings

### PR DESCRIPTION
**Pull request by bgyori**
The line ```from pkg_resources import DistributionNotFound, get_distribution``` produces deprecation warnings (see https://setuptools.pypa.io/en/latest/pkg_resources.html). This PR implements a variant for Python 3.8+ that avoids these deprecations.

### CODE REVIEW

File: obonet/__init__.py
Comment (dhimmel) [inline obonet/__init__.py:26]: Thanks for the contribution!

What do you think about bumping to 3.8 at

https://github.com/dhimmel/obonet/blob/fec6a82e53d01338c85e77039a4dc05288f6ab99/pyproject.toml#L31

and then removing the `from pkg_resources import DistributionNotFound, get_distribution` approach altogether?

Comment (dhimmel) [reply obonet/__init__.py:26]: 3.7 end of life as of June 27, 2023 and pip should know to install an older release of obonet if using python 3.7

Comment (bgyori) [reply obonet/__init__.py:26]: That's fine with me, I tend to always err on the side of broader compatibility unless there is a meaningful feature being blocked by it, but I think in this case, this would not be a big issue.

Comment (dhimmel) [reply obonet/__init__.py:26]: Cool. I'm happy to make the changes after merging this or you can. Let me know which you prefer!

Comment (bgyori) [reply obonet/__init__.py]: I made the changes, feel free to make further modifications if needed.

**Review by dhimmel — APPROVED** [review]
Thanks! I'll drop a note here once there's a release.

**Comment by dhimmel** [comment]
Release `1.1.0` now on [PyPI](https://pypi.org/project/obonet/1.1.0/) now contains the merged commit.

**Pull request merged**
